File: TypeScript/src/frogger/froggerid.ts
/**
 * Refactor Task 2. Can we use this somewhere?
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22), Duan Liang (F23)
 */
class FroggerID {
	firstName: string;
	lastName: string;
	phoneNumber: string;
	zipCode: string;
	state: string;
	gender: string

	constructor(firstName: string, lastName: string, phoneNumber: string, zipCode: string, state: string, gender: string) {
		this.firstName = firstName;
		this.lastName = lastName;
		this.phoneNumber = phoneNumber;
		this.zipCode = zipCode;
		this.state = state;
		this.gender = gender;
	}

	public equals(other: FroggerID): boolean {
		return this.firstName === other.firstName && this.lastName === other.lastName && this.phoneNumber === other.phoneNumber
			&& this.zipCode === other.zipCode && this.state === other.state && this.gender === other.gender;
	}
}

File: Java/src/main/java/frogger/FroggerID.java
package frogger;

// A record class that can conveniently store immutable data. 
// To compare two record classes with their corresponding fields, simply use ".equals()".
// More information: https://docs.oracle.com/en/java/javase/17/language/records.html#GUID-6699E26F-4A9B-4393-A08B-1E47D4B2D263
public record FroggerID(String firstName,
                     String lastName,
                     String phoneNumber,
                     String zipCode,
                     String state,
                     String gender) {
}